### ISSUE TITLE
Dynamic routes are incorrectly cached

### ISSUE THREAD

**Issue by zupat** [codemicro/fiber-cache#3]
I don't know if it was intended, but this package doesn't work for dynamic routes.

A route defined as "/users/:userID" will yield the same cached response for "/users/123" and "/users/456" since the key is not dynamic. This can be verified by adding and running this test.

```
func Test_DynamicRoutes(t *testing.T) {
	createNewCacheForTest()
	app := fiber.New()

	responseText1 := "thisIsAResponse"
	responseText2 := "thisIsADifferentResponse"

	app.Get("/users/:userID", New(), func(c *fiber.Ctx) error {
		c.SendString(responseText1 + c.Params("userID", "defaultUser"))
		return nil
	})

	resp1, _ := app.Test(httptest.NewRequest("GET", "/users/123", nil))
	resp2, _ := app.Test(httptest.NewRequest("GET", "/users/456", nil))

	returnedResponse1 := getResponseBody(resp1)
	returnedResponse2 := getResponseBody(resp2)

	if returnedResponse1 == returnedResponse2 {
		t.Fatal("Cache returned the same response for different urls")
	}
```

**Comment by zupat** [comment]
This can be solved by using the request path for the key (c.Path()) and possibly concatenating the first item in the "Accept" request header(so different responses will be cache for the same path based on the type that the user requested)

**Comment by codemicro** [comment]
Hi @jinyus, thanks for the issue!

I was aware of the lack of dynamic route support but I had no idea what to do about it, since I didn't even know `c.Path` was a thing until last night. I was planning to address that in this middleware at some point today, which I'll still do.

I am not going to add anything dealing with the `Accept` (or any other) headers into the middleware, just because of how difficult it is to determine what should be cached/sent when and in what scenarios.

**Comment by codemicro** [comment]
This has been fixed in [v2.1.1](https://github.com/codemicro/fiber-cache/releases/tag/v2.1.1). Thanks again for the PR!
